scaling_performance/optimizing_storage.adoc

```diff
@@ -11,25 +11,185 @@
+| CNS/CRS GlusterFS footnoteref:[disclaimer], Ceph Object Storage (RADOS Gateway), OpenStack Swift, Aliyun OSS, AWS S3, Google Cloud Storage, Azure Blob Storage, Vendor S3 footnoteref:[disclaimer3], Vendor Swift footnoteref:[disclaimer3]
+|===
+
+[NOTE]
+====
+As of {product-title} 3.6.1, Container-Native Storage (CNS) GlusterFS (a hyperconverged or cluster-hosted storage solution) and Container-Ready Storage (CRS)
+GlusterFS (an externally hosted storage solution) provides interfaces for block, file, and object storage for the purpose of the {product-title} registry, logging, and metrics.
+====
+
+[[back-end-recommendations]]
+=== Storage Recommendations
+
+The following table summarizes the recommended and configurable storage technology for the given {product-title} cluster application.
+
+.Recommended and configurable storage technology
+[options="header"]
+|===
+|Storage type |ROX |RWX |Registry|Scaled registry| Metrics|Logging|Apps
+
+| Block
+| Yes footnoteref:[disclaimer4, This does not apply to physical disk, VM physical disk, VMDK, loopback over NFS, AWS EBS, and Azure Disk.]
+| No
+| Configurable
+| Not configurable
+| Recommended
+| Recommended
+| Recommended
+
+| File
+| Yes footnoteref:[disclaimer4]
+| Yes
+| Configurable
+| Configurable
+| Configurable
+| Configurable
+| Recommended
+
+| Object
+| Yes
+| Yes
+| Recommended
+| Recommended
+| Not configurable
+| Not configurable
+| Not configurable footnoteref:[disclaimer5, Object storage is not consumed through {product-title}'s PVs/persistent volume claims (PVCs). Apps must integrate with the object storage REST API.]
+|===
+
+[NOTE]
+====
+A scaled registry is an {product-title} registry where 3 or more pod replicas are running.
+====
+
+[[application-storage-recommendations]]
+==== Specific Application and Storage Recommendations
+
+[[registry]]
+===== Registry
+
+In a non-scaled/high-availability (HA) {product-title} registry cluster deployment:
+
+* The preferred storage technology is object storage followed by block storage. The
+storage technology does not need to support ReadWriteMany (RWX) access mode.
+* The storage technology must ensure read-after-write consistency. All NAS storage (excluding CNS/CRS GlusterFS as it uses an object storage interface) are not
+recommended for {product-title} Registry cluster deployment with production workloads.
+* While hostPath volumes are configurable for a non-scaled/HA {product-title} Registry cluster deployment, they are not recommended.
+
+[WARNING]
+====
+Corruption may occur when using NFS to back {product-title} registry at production workloads.
+====
+
+[[scaled-registry]]
+===== Scaled Registry
+
+In a scaled/HA {product-title} registry cluster deployment:
+
+* The preferred storage technology is object storage. The storage technology must support RWX access mode and must ensure read-after-write consistency.
+* File storage and block storage are expressly not recommended for a scaled/HA {product-title} registry
```

Comment:
'expressly' isn't needed.  It's 'not recommended' is strong enough.

Reply:
I chose the word 'expressly' since the current documentation is so weak on the topic [0][1][2].  I'd like to link to this document from those docs once this lands.


[0] https://docs.openshift.com/container-platform/3.6/install_config/registry/deploy_registry_existing_clusters.html#registry-production-use
[1] https://docs.openshift.com/container-platform/3.6/install_config/aggregate_logging.html#aggregated-elasticsearch
[2] https://docs.openshift.com/container-platform/3.6/install_config/cluster_metrics.html#capacity-planning-for-openshift-metrics